online-bookstore/online-bookstore/src/main/java/com/example/online_bookstore/implementation/BookOrderServiceRepo.java

```diff
@@ -0,0 +1,67 @@
+package com.example.online_bookstore.implementation;
+
+import com.example.online_bookstore.models.BookOrder;
+import com.example.online_bookstore.repository.BookOrderRepo;
+import com.example.online_bookstore.service.BookOrderService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+
+@Component
+public class BookOrderServiceRepo implements BookOrderService {
+
+    @Autowired
+    BookOrderRepo repo;
+
+    @Override
+    public List<BookOrder> findBookOrderAfterDate(Date date) {
+        return repo.findByStartDateAfter(date);
+    }
+
+    @Override
+    public List<BookOrder> findBookOrderBeforeDate(Date date) {
+        return repo.findByStartDateBefore(date);
+    }
+
+    @Override
+    public List<BookOrder> findBookOrderBetweenDates(Date startDate, Date endDate) {
+        return repo.findByStartDateBetween(startDate, endDate);
+    }
+
+    @Override
+    public List<BookOrder> findCompletedBookOrder(Date date) {
+        return repo.findByOrderFinishedDateBeforeAndOrderFinishedDateIsNotNull(date);
+    }
```

Comment:
this name is too long and there is no need to have this kind of description in the name of the method. 
The way method works can be explained in javadoc. Use javadoc in Service (not ServiceImpl) to describe what method is supposed to do.

Reply:
Rename this to sth like findCompleteBookOrder